src/test/kotlin/com/pubnub/contract/membership/step/WhenSteps.kt

```diff
@@ -0,0 +1,100 @@
+package com.pubnub.contract.membership.step
+
+import com.pubnub.api.models.consumer.objects.membership.PNChannelDetailsLevel
+import com.pubnub.api.models.consumer.objects.membership.PNChannelMembership
+import com.pubnub.contract.state.World
+import com.pubnub.contract.uuidmetadata.state.MembershipState
+import com.pubnub.contract.uuidmetadata.state.UUIDMetadataState
+import io.cucumber.java.en.When
+
+class WhenSteps(
+    private val world: World,
+    private val membershipState: MembershipState,
+    private val uuidMetadataState: UUIDMetadataState
+) {
+    @When("I get the memberships")
+    fun i_get_the_memberships() {
+        world.pubnub.getMemberships(uuid = uuidMetadataState.uuid).sync()?.let {
```

Comment:
Do we want to mix membershipState and uuidMetadataState here?
Wouldn't it be better to operate in one test on one state object?

Reply:
I don't know what would be better. To have just one type of state I think I should combine all `objectsv2` states into one and I'm not sure if I should do it like that or not. I guess sooner or later this idea of separate state objects will fall as soon as we start to actually implement more features. For example for now we have step: `And the id for 'Amelia' persona` which should store the `id` of a user somewhere, but it's so universal step that I believe we could use it in so many other features.